streams/src/main/java/org/apache/kafka/streams/processor/DefaultPartitionGrouper.java

```diff
@@ -29,6 +29,12 @@
 import java.util.Map;
 import java.util.Set;
 
+/**
+ * DefaultPartitionGrouper groups partitions by the partition id. This behavior is assumed by the join processing in KStream.
+ * Join processing requires that topics are copartitoned, i.e., being partitioned by the same key and having the same by
```

Comment:
"same by number of partitions" remove "by", and also this sentence is a bit wired as "requires topics are copartitioned .. are grouped together". Did you mean "require topics are co-partitioned, i.e.... so that partitions with the same id can be grouped together"?